fuzzers/eclipser/fuzzer.py

```diff
@@ -11,88 +11,181 @@
+    # Reset compilation flags only using NO_SANITIZER_COMPAT_* flags. Eclipser
+    # module prefers unoptimized binaries, as hacky binary code can impede its
+    # grey-box concolic technique. Note that we can still use optimized binaries
+    # for AFL module.
+    new_env['CFLAGS'] = ' '.join(utils.NO_SANITIZER_COMPAT_CFLAGS)
+    cxxflags = [utils.LIBCPLUSPLUS_FLAG] + utils.NO_SANITIZER_COMPAT_CFLAGS
+    new_env['CXXFLAGS'] = ' '.join(cxxflags)
+    uninstrumented_outdir = get_uninstrumented_outdir(os.environ['OUT'])
+    os.mkdir(uninstrumented_outdir)
+    new_env['OUT'] = uninstrumented_outdir
+    fuzz_target = os.getenv('FUZZ_TARGET')
+    if fuzz_target:
+        targ_name = os.path.basename(fuzz_target)
+        new_env['FUZZ_TARGET'] = os.path.join(uninstrumented_outdir, targ_name)
+    print('[build] Re-building benchmark for uninstrumented fuzzing target')
+    utils.build_benchmark(env=new_env)
 
-def fuzz(input_corpus, output_corpus, target_binary):
-    """Run fuzzer."""
-    # Create an encoded temp corpus directory.
-    encoded_temp_corpus = os.path.join(os.path.dirname(input_corpus),
-                                       'temp-corpus')
-    if not os.path.exists(encoded_temp_corpus):
-        os.mkdir(encoded_temp_corpus)
 
-    print('[fuzz] Running target with Eclipser')
+def eclipser(input_corpus, output_corpus, target_binary):
+    """Run Eclipser."""
+    # We will use output_corpus as a directory where AFL and Eclipser sync their
+    # test cases with each other. For Eclipser, we should explicitly specify an
+    # output directory under this sync directory.
+    eclipser_out = os.path.join(output_corpus, "eclipser_output")
     command = [
         'dotnet',
         '/Eclipser/build/Eclipser.dll',
-        'fuzz',
         '-p',
         target_binary,
-        '-t',
-        '1048576',  # FIXME: Find the max value allowed here.
+        '-s',
+        output_corpus,
         '-o',
-        encoded_temp_corpus,
-        '--src',
-        'file',
-        '--initarg',
-        'foo',  # Specifies how command line argument is passed, just a file.
-        '-f',
+        eclipser_out,
+        '--arg',  # Specifies the command-line of the program.
+        'foo',
+        '-f',  # Specifies the path of file input to fuzz.
         'foo',
-        '--maxfilelen',
-        # Default is too low (8 bytes), match experiment config at:
-        # https://github.com/SoftSec-KAIST/Eclipser-Artifact/blob/6aadf02eeadb0416bd4c5edeafc8627bc24ebc82/docker-scripts/experiment-scripts/package-exp/run_eclipser.sh#L25
-        '1048576',
-        # Default is low (0.5 sec), recommended to use higher:
-        # https://github.com/google/fuzzbench/issues/70#issuecomment-596060572
+        '-v',  # Controls the verbosity.
+        '2',
         '--exectimeout',
         '5000',
     ]
-    if os.listdir(input_corpus):  # Important, otherwise Eclipser crashes.
+    if os.listdir(input_corpus):  # Specify inputs only if any seed exists.
```

Comment:
Can you please add support for crash dir, similar to this
https://github.com/google/fuzzbench/blob/master/fuzzers/libfuzzer/fuzzer.py#L52
we are working on crash based benchmarking, so this will be great to add as you are modifying this.

Reply:
In case it's not clear what this comment means, we want any crashes eclipser finds to be located in `output_corpus`.

Reply:
Is there any specific requirement (or protocol) for the directory path for crashes?

Currently, Eclipser uses a similar path convention with AFL. As a result, eclipser/fuzzer.py will store the found crashes in
`<output_corpus>/afl-master/crashes`, `<output_corpus>/afl-worker/crashes`, and `<output_corpus>/eclipser-output/crashes`. 

Is this behavior okay?

Reply:
Ah perfect, in afl's case, nothing is needed to be changed, it should just work.

Reply:
Yes. It is fine.